src/lib.rs

```diff
@@ -1172,3 +1172,131 @@ pub fn set_overwrite(json: &str, path: &str, value: &str) -> Result<String, GJSO
 }
+
+#[inline]
+pub fn get_with_key<'a>(json: &'a str, path: &'a str) -> (Value<'a>, String) {
+    // This is a duplicate of gjson's get() method, but it returns the json key
+    // as the second value in the tuple.
+    // For example, if we give path `object.user.email`, the returned key will be `email`.
+    let mut path = path;
+    let mut lines = false;
+    if path.len() >= 2 && path.as_bytes()[0] == b'.' && path.as_bytes()[1] == b'.' {
+        // json lines
+        path = tostr(&path.as_bytes()[2..]);
+        lines = true;
+    }
+    let path = Path::new(path);
+    let (res, path) = {
+        let json = json.as_bytes();
+        if lines {
+            let res = get_arr(json, 0, true, path);
+            (res.0, res.2)
+        } else if path.is_modifier() {
+            modifiers::exec(json, path)
+        } else if path.is_multipath() {
+            multipath::exec(json, path)
+        } else {
+            let mut i = 0;
+            loop {
+                if i == json.len() {
+                    break (Value::default(), path);
+                }
+                if json[i] <= b' ' {
+                    i += 1;
+                    continue;
+                }
+                if json[i] == b'{' {
+                    let res = get_obj(json, i, path);
+                    break (res.0, res.2);
+                }
+                if json[i] == b'[' {
+                    let res = get_arr(json, i, false, path);
+                    break (res.0, res.2);
+                }
+                break (Value::default(), path);
+            }
+        }
+    };
+    if !path.more() {
+        let p = String::from_utf8(path.comp.to_vec()).expect("Found invalid UTF-8");
+        return (res, p);
+    }
+    let sub_path = tostr(path.extra);
+    let mut json = if res.slice.len() > 0 {
+        crate::get(&res.slice, sub_path)
+    } else {
+        json_into_owned(crate::get(&res.owned, sub_path))
+    };
+    let mut index = None;
+    if let Some(index1) = res.index {
+        if let Some(index2) = json.index {
+            index = Some(index1 + index2);
+        }
+    }
+    json.index = index;
+
+    let p = String::from_utf8(path.comp.to_vec()).expect("Found invalid UTF-8");
+    (json, p)
+}
+
+#[inline]
+pub fn delete_path<'a>(json: &'a str, path: &'a str) -> Result<String, GJSONError> {
+    if !valid(json) {
+        return Err(GJSONError { msg: "invalid json".to_string() })
+    }
+
+    let (existing_value, key) = get_with_key(json, path);
+
+    if !existing_value.exists() {
+        return Err(GJSONError { msg: "path does not exist".to_string() })
+    }
+
+    // Somehow find where the key starts and the value ends
+    let value_begin = existing_value.index().unwrap();
+
+    let key_start = match find_key_start(value_begin, key, json.as_bytes()) {
+        Some(x) => x,
+        None => {
+            return Err(GJSONError { msg: "could not find key".to_string() })
```

Comment:
If the path is not found, the "could not find key" err is bubbled up to the SDK?

Reply:
Yes, it will return the error in the wasm response to the SDK